### PR TITLE
feat: Add Analytics for attach images - MEED-2435 - Meeds-io/MIPs#53

### PR DESCRIPTION
**Pull request by SaraBoutej**
This PR allows to add information about the module and the sub module for attach image analytics

### CODE REVIEW

**Comment by boubaker** [comment]
No Addons to add in analytics please. In fact, this addon is in the bottom layer of addons just after Social.

File: analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/BaseAttachmentAnalyticsListener.java [hunk 1]
```diff
@@ -72,29 +75,40 @@ public void onEvent(Event<String, ObjectAttachmentId> event) throws Exception {
...
   }
 
-  private void buildStatisticData(String operation, long spaceId, long userId) {
+  private StatisticData buildStatisticData(String operation, ObjectAttachmentId objectAttachment, long spaceId, long userId) {
     Space space = spaceService.getSpaceById(String.valueOf(spaceId));
     StatisticData statisticData = new StatisticData();
-    statisticData.setModule("social");
-    statisticData.setSubModule("attachment");
+    statisticData.setModule(getModule());
+    statisticData.setSubModule(getSubModule());
```
Comment (boubaker) [inline analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/BaseAttachmentAnalyticsListener.java]: Add `objectAttachment` in params of method as shown to let upper layer listeners be able to determine its value dynamically.

```suggestion
    statisticData.setModule(getModule(objectAttachment));
    statisticData.setSubModule(getSubModule(objectAttachment));
```

Comment (SaraBoutej) [reply analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/BaseAttachmentAnalyticsListener.java]: done!

File: analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/ActivityAttachmentAnalyticsListener.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  protected void extendStatisticData(StatisticData statisticData, ObjectAttachmentId objectAttachment) {
+    ExoSocialActivity activity = activityManager.getActivity(objectAttachment.getObjectId());
+    if (activity != null) {
+      String objectType;
+      if (activity.getType() == null) {
+        objectType = activity.isComment() ? "activityComment" : "activity";
+      } else {
+        objectType = activity.getType();
+      }
+      statisticData.addParameter("activityType", objectType);
```
Comment (boubaker) [inline analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/ActivityAttachmentAnalyticsListener.java]: Please move **a part** of `org.exoplatform.analytics.listener.social.AnalyticsActivityListener.addActivityStatisticEvent` to `org.exoplatform.analytics.utils.AnalyticsUtils` in order to reuse the same collected statistics data for activities & comments independing from the operation made.

```suggestion
      addActivityStatistics(statisticsData, activity);
```

(Similar to `org.exoplatform.analytics.utils.AnalyticsUtils.addSpaceStatistics(StatisticData, Space)`)

Comment (SaraBoutej) [reply analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/ActivityAttachmentAnalyticsListener.java]: done

File: analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/ActivityAttachmentAnalyticsListener.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                                             ActivityManager activityManager,
+                                             InitParams initParam) {
+    super(attachmentService, spaceService, initParam);
+    this.activityManager = activityManager;
+  }
+
+  @Override
+  protected void extendStatisticData(StatisticData statisticData, ObjectAttachmentId objectAttachment) {
+    ExoSocialActivity activity = activityManager.getActivity(objectAttachment.getObjectId());
+    if (activity != null ) {
```
Comment (boubaker) [inline analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/ActivityAttachmentAnalyticsListener.java]:
```suggestion
    if (activity != null) {
```

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: analytics-webapps/src/main/resources/locale/portlet/Analytics_en.properties [hunk 4]
```diff
@@ -394,6 +394,7 @@ analytics.mousedown=Click Mouse down
 analytics.mouseup=Click Mouse up
 analytics.attachImages=Attach images
 analytics.removeImageAttachments=Remove image attachments
+analytics.activityComment=Activity comment
```
Comment (boubaker) [inline analytics-webapps/src/main/resources/locale/portlet/Analytics_en.properties]: Not used anymore

```suggestion
```

Comment (SaraBoutej) [reply analytics-webapps/src/main/resources/locale/portlet/Analytics_en.properties]: done

**Review by boubaker — APPROVED** [review]

**Pull request merged**